### PR TITLE
Fix: [None] Add notification_preferences and candidate_acceptances to database subset

### PR DESCRIPTION
**Pull request by mganderson**
This PR adds the `notification_preferences` and `candidate_acceptances` tables to the database subset.

The records that are included from those tables correspond to `user` and `candidate` records previously included in the subset -- that is, about one in thirty records are captured from the `notification_preferences` and `candidate_acceptances` tables.

Comments regarding this fix are made on the diff.

### CODE REVIEW

File: config.ripplematch.json [hunk 1]
```diff
@@ -12,6 +12,14 @@
       "table": "public.application",
       "where": "student_id IS NOT NULL AND MOD(student_id, 30) = 0"
     },
+    {
+      "table": "public.notification_preferences",
+      "where": "user_id in (SELECT id FROM public.user WHERE recruiter_id IS NOT NULL OR MOD(student_id, 30) = 0)"
```
Comment (mganderson) [inline config.ripplematch.json:17]: This is a bit hacky, but the "where" condition uses a subquery so that we select records corresponding to the users we selected at lines 4-5.

File: config.ripplematch.json [hunk 2]
```diff
@@ -12,6 +12,14 @@
...
+    },
+    {
+      "table": "public.candidate_acceptances",
+      "where": "candidate_id IN (SELECT id FROM public.candidate WHERE student_id IS NOT NULL AND MOD(student_id, 30) = 0)"
```
Comment (mganderson) [inline config.ripplematch.json:21]: Similarly, we want only candidate_acceptance records associated with the one-in-thirty candidate records we captured at lines 8-9.

File: config.ripplematch.json [hunk 3]
```diff
@@ -837,7 +845,28 @@
     }
   ],
   "fk_augmentation": [],
-  "upstream_filters": [],
+  "upstream_filters": [
```
Comment (mganderson) [inline config.ripplematch.json:848]: I'd be lying if I said I fully understood the issue I was having before, but adding this conditions to "upstream_filters" resolves the problem I was having.  My rough idea of what was happening is as follows:

Condenser works by creating a directed graph of database tables where foreign keys are the edges between nodes.  The starting nodes are the tables included in `initial_targets`.  Condenser starts at those initial_target nodes, then traverses the graph, grabbing the minimum records from each table it visits necessary to maintain referential integrity with the initial_target tables.

Without listing the _same tables in `initial_targets`_ in the `"upstream_filters"` field along with a filter condition that would never evaluate to true, the script would hang at certain points when I first tried to add the `notification_preferences` and `candidate_acceptances` tables.  What I _think_ was going on is that the the condenser script was being forced to revisit the `user` table by virtue of `notification_preferences`'s foreign key to `user` (and the `user` table had already been subset, leading to... issues), and something similar was happening for `candidate_acceptance`-> `candidate`, `role`, or `recruiter`.  

Basically, we're saying to condenser: don't visit these tables again; we already captured them in `"initial_targets"`.  I think.

**Review by DanielBalsam — APPROVED** [review]
nice!
